### PR TITLE
More guidance for using the Demos repo

### PR DESCRIPTION
**Pull request by mikehoffms**
Sibling Demos repo PR:  https://github.com/MicrosoftEdge/Demos/pull/14 - "Edit pass on topmost Readme"

Changes made in this Docs repo PR:

* Improved article "Sample code for DevTools":
   * Added h2 section "List of samples".
   * Added h2 section "To Do demo".
   * Added h2 section "Download or clone the Demos repo". 
   * Added h2 section "Open a demo folder in the Sources tool and edit a file".
   * Added h2 section "Open a demo folder in Visual Studio Code".
   * https://review.learn.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/sample-code/sample-code?branch=pr-en-us-2430 [[gh rendered](https://github.com/mikehoffms/edge-developer/blob/user/mikehoffms/clone-demos/microsoft-edge/devtools-guide-chromium/sample-code/sample-code.md)]

* Added a Samples page to the **Microsoft Edge extensions** TOC bucket, corresponding to that section of the [Demos repo's topmost Readme](https://github.com/MicrosoftEdge/Demos#readme).
   *  https://review.learn.microsoft.com/en-us/microsoft-edge/extensions-chromium/samples?branch=pr-en-us-2430 [[gh rendered](https://github.com/mikehoffms/edge-developer/blob/user/mikehoffms/clone-demos/microsoft-edge/extensions-chromium/samples.md)]

* In the "Sources tool" article, applied h2/h4/h6 pattern, and linked to the new h2 "Open a demo folder in the Sources tool and edit a file".
   * https://review.learn.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/sources/?branch=pr-en-us-2430#using-the-editor-pane-to-view-or-edit-files [[gh rendered](https://github.com/mikehoffms/edge-developer/blob/user/mikehoffms/clone-demos/microsoft-edge/devtools-guide-chromium/sources/index.md#using-the-editor-pane-to-view-or-edit-files)]

* Cross-linked WebView2, DevTools, DevTools extension for VS Code.
   * "Download the WebView2Samples repo" in _Set up your Dev environment for WebView2_:
      * https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/how-to/machine-setup?branch=pr-en-us-2430#download-the-webview2samples-repo [[gh rendered](https://github.com/mikehoffms/edge-developer/blob/user/mikehoffms/clone-demos/microsoft-edge/webview2/how-to/machine-setup.md#download-the-webview2samples-repo)]
   * "Step 5: Clone the Demos repo" in _Installing the DevTools extension for Visual Studio Code_:
      * https://review.learn.microsoft.com/en-us/microsoft-edge/visual-studio-code/microsoft-edge-devtools-extension/install?branch=pr-en-us-2430#step-5-clone-the-demos-repo [[gh rendered](https://github.com/mikehoffms/edge-developer/blob/user/mikehoffms/clone-demos/microsoft-edge/visual-studio-code/microsoft-edge-devtools-extension/install.md#step-5-clone-the-demos-repo)]

AB#43114410

### CODE REVIEW

File: microsoft-edge/devtools-guide-chromium/sample-code/sample-code.md [hunk 1]
```diff
@@ -90,13 +144,87 @@ This is the source code repo and its directory which stores the files for this d
...
+
+   ![Prompted to open the cloned repo](./sample-code-images/prompt-open-cloned-repo.png)
+
+1. If prompted **Do you trust...**, click the **Yes** button.  Or, click the **No** button and continue with most parts of this walkthrough.
+
+   The **Explorer** tree lists many demos, including **demo-to-do**.
+
+1. In Visual Studio Code, select **File** > **Close Folder**.
+
+1. In the Activity Bar, click the **Microsoft Edge Tools** (![Microsoft Edge Tools icon](./sample-code-images/microsoft-edge-tools-icon.png)) button.  The **Microsoft Edge Tools** pane opens.
```
Comment (captainbrosset) [inline microsoft-edge/devtools-guide-chromium/sample-code/sample-code.md:204]: I may have missed it, but I don't think we're telling people to install the Microsoft Edge Tools extension before. So they won't have this icon in the Activity bar.

Also, this section is about cloning the repo, not using the Edge tools in VS Code. So I don't think this part is necessary.

File: microsoft-edge/web-platform/samples.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+---
```
Comment (captainbrosset) [inline microsoft-edge/web-platform/samples.md:1]: I'm not sure if we need this page at all.

None of the web platform APIs that are demo'd on the repo are documented or used on LMC.
These demo pages were not created for LMC, but instead for blog posts or conference talks.

**Review by captainbrosset — CHANGES_REQUESTED** [review]

File: microsoft-edge/web-platform/samples.md (hunk 2 above)
Comment (mikehoffms) [reply microsoft-edge/web-platform/samples.md:1]: removed web-platform/samples.md

File: microsoft-edge/devtools-guide-chromium/sample-code/sample-code.md (hunk 1 above)
Comment (mikehoffms) [reply microsoft-edge/devtools-guide-chromium/sample-code/sample-code.md]: reworked and expanded to give more guidance on what you can to with the local Demos repo to open and edit files.  Body doesn't mention the extension now.  See also links, instead.

File: microsoft-edge/extensions-chromium/samples.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+author: MSEdgeTeam
+ms.author: msedgedevrel
+ms.topic: conceptual
+ms.prod: microsoft-edge
+ms.technology: devtools
+ms.date: 01/27/2023
+---
+# Samples for Microsoft Edge extensions
+
+Use the demos in the [MicrosoftEdge / Demos](https://github.com/MicrosoftEdge/Demos#readme) repo to learn how to use Microsoft Edge to develop webpages and web apps.
```
Comment (captainbrosset) [inline microsoft-edge/extensions-chromium/samples.md:13]:
```suggestion
Use the demos in the [MicrosoftEdge / Demos](https://github.com/MicrosoftEdge/Demos#readme) repo to learn how to use Microsoft Edge to develop extensions.
```

I'll leave that up to you, but this section of our TOC is specifically about developing browser extensions, not webpages or web apps. So this paragraph should reflect this.

File: microsoft-edge/visual-studio-code/microsoft-edge-devtools-extension/install.md [hunk 4]
```diff
@@ -235,3 +235,7 @@ Next, see [Get started using the DevTools extension for Visual Studio Code](./ge
 * [Get started using the DevTools extension for Visual Studio Code](./get-started.md)
 * [Opening DevTools and the DevTools browser](./open-devtools-and-embedded-browser.md)
 * [Microsoft Edge DevTools extension for Visual Studio Code](../microsoft-edge-devtools-extension.md)
+* [Sample code for DevTools](../../devtools-guide-chromium/sample-code/sample-code.md)
+* [Download the WebView2Samples repo](../../webview2/how-to/machine-setup.md#download-the-webview2samples-repo) in _Set up your Dev environment for WebView2_.
+   * [Clone the WebView2Samples repo](../../webview2/how-to/machine-setup.md#clone-the-webview2samples-repo) in _Set up your Dev environment for WebView2_.
```
Comment (captainbrosset) [inline microsoft-edge/visual-studio-code/microsoft-edge-devtools-extension/install.md:240]: How come we talk about WebView2 in this VS Code DevTools extension doc? I don't see the link here.

File: microsoft-edge/devtools-guide-chromium/sample-code/sample-code.md [hunk 5]
```diff
@@ -5,19 +5,85 @@ author: MSEdgeTeam
...
+| My tasks/TODO app | A simple To Do app used in various DevTools documentation. | [demo-to-do](https://github.com/MicrosoftEdge/Demos/tree/main/demo-to-do) |
+| Detached elements | A chat-like demo with DOM memory leaks showing how to use the **Detached Elements** tool. | [detached-elements](https://github.com/MicrosoftEdge/Demos/tree/main/detached-elements) |
+| 3D View demo | Demonstrates the **3D View** tool. | [devtools-3d](https://github.com/MicrosoftEdge/Demos/tree/main/devtools-3d) |
+| Animal shelter/Accessibility issues | Demonstrates features to detect accessibility issues. | [devtools-a11y-testing](https://github.com/MicrosoftEdge/Demos/tree/main/devtools-a11y-testing) |
+| Console panel demo pages | 15 webpages demonstrating **Console** features. | [devtools-console](https://github.com/MicrosoftEdge/Demos/tree/main/devtools-console) |
+| Testing all badges in DevTools for contrast issues | Used by a blog post about improving contrast using DevTools. | [devtools-contrast-bugfix](https://github.com/MicrosoftEdge/Demos/tree/main/devtools-contrast-bugfix) |
```
[the remark this answers is not in the document]
Comment (mikehoffms) [reply microsoft-edge/devtools-guide-chromium/sample-code/sample-code.md:48]: yes b/c it is a sample about DevTools, so seems to fit criteria for this table. The scope of table isn't regular docs, but rather, all samples that are about DevTools.
You might know about this particular demo.  I'm inclined to keep.

**Review by JasonAndrewWriter — APPROVED** [review]

File: microsoft-edge/visual-studio-code/microsoft-edge-devtools-extension/install.md (hunk 4 above)
Comment (mikehoffms) [reply microsoft-edge/visual-studio-code/microsoft-edge-devtools-extension/install.md:240]: removed wv2 lines

File: microsoft-edge/extensions-chromium/samples.md (hunk 3 above)
Comment (mikehoffms) [reply microsoft-edge/extensions-chromium/samples.md:13]: Use the extensions demos in the [Demos](https://github.com/MicrosoftEdge/Demos#readme) repo to learn how to develop extensions for Microsoft Edge.

**Pull request merged**